### PR TITLE
Move more details into sub generators

### PR DESCRIPTION
**Pull request by thomasvl**
This should get most of the last field specific knowledge out of the MessageGenerator and into the sub Generators.  This leaves the (Field,Storage,Oneof)Generators for the final cleanup of roles/knowledge/use of Descriptors.

### CODE REVIEW

File: Sources/PluginLibrary/SwiftProtobufInfo.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/// Scope for helpers about the library.
+public enum SwiftProtobufInfo {
+
+  /// Proto Files that ship with the library.
+  public static let bundledProtoFiles: Set<String> = [
+    "google/protobuf/any.proto",
+    "google/protobuf/api.proto",
+    // "google/protobuf/descriptor.proto"
+    //    is not a WKT; and while it is used by generator, it does NOT ship
+    //    with the library, if the library every wants to expose full
```
Comment (allevato) [inline Sources/PluginLibrary/SwiftProtobufInfo.swift]: s/every/ever/

File: Sources/protoc-gen-swift/Descriptor+Extensions.swift [hunk 2]
```diff
@@ -64,6 +75,28 @@ extension Descriptor {
...
+
+  /// A `String` containing a comma-delimited list of Swift range expressions
+  /// covering the extension ranges for this message.
+  ///
+  /// This expression list is suitable as a pattern match in a `case`
+  /// statement. For example, `"case 5..<10, 20..<30:"`.
+  var swiftExtensionRangeExpressions: String {
+    return extensionRanges.lazy.map {
+      $0.swiftRangeExpression
+      }.joined(separator: ", ")
```
Comment (allevato) [inline Sources/protoc-gen-swift/Descriptor+Extensions.swift]: Dedent

File: Sources/protoc-gen-swift/Descriptor+Extensions.swift [hunk 3]
```diff
@@ -64,6 +75,28 @@ extension Descriptor {
...
+
+  /// A `String` containing a Swift Boolean expression that tests if the given
+  /// variable is in any of the extension ranges for this message.
+  ///
+  /// - Parameter variable: The name of the variable to test in the expression.
+  /// - Returns: A `String` containing the Boolean expression.
+  func swiftExtensionRangeBooleanExpression(variable: String) -> String {
+    return extensionRanges.lazy.map {
+      "(\($0.swiftBooleanExpression(variable: variable)))"
+      }.joined(separator: " || ")
```
Comment (allevato) [inline Sources/protoc-gen-swift/Descriptor+Extensions.swift]: Dedent

File: Sources/PluginLibrary/SwiftProtobufInfo.swift (hunk 1 above)
Comment (thomasvl) [inline Sources/PluginLibrary/SwiftProtobufInfo.swift]: Done

File: Sources/protoc-gen-swift/Descriptor+Extensions.swift (hunk 2 above)
Comment (thomasvl) [inline Sources/protoc-gen-swift/Descriptor+Extensions.swift]: Done

File: Sources/protoc-gen-swift/Descriptor+Extensions.swift (hunk 3 above)
Comment (thomasvl) [inline Sources/protoc-gen-swift/Descriptor+Extensions.swift]: Done

**Pull request merged**